homeassistant/components/mqtt/__init__.py

```diff
@@ -719,7 +740,7 @@ async def async_disconnect(self):
 
         def stop():
             """Stop the MQTT client."""
-            self._mqttc.disconnect()
+            # Do not disconnect, we want the broker to always publish will
```

Comment:
Shouldn't HomeAssistant set the `DEFAULT_PAYLOAD_NOT_AVAILABLE` explicitly and then disconnect gracefully here? Otherwise, it may take a while until the MQTT broker detects that the client disappeared, leading to an unnecessary delay...

Reply:
The connection to the MQTT server is actively broken, and the broker will immediately publish the will.

What you describe is an issue if the connection is not actively broken, in which case the TCP connection will linger until it times out.

If you think this is not the case, please provide a way to reproduce the issue you suggest.